src/main/java/domain/lotto/result/LottoResult.java

```diff
@@ -0,0 +1,28 @@
+package domain.lotto.result;
+
+import static java.util.stream.Collectors.*;
+
+import java.util.List;
+import java.util.Objects;
+
+import domain.store.WinningMoney;
+import domain.lotto.rank.LottoRank;
+
+public class LottoResult {
+	private List<LottoRank> lottoRanks;
+
+	public LottoResult(List<LottoRank> lottoRanks) {
+		this.lottoRanks = lottoRanks;
+	}
+
+	public WinningMoney calculateWinning() {
+		return lottoRanks.stream()
+			.reduce(new WinningMoney(0), ((total, lottoRank) -> lottoRank.addWinning(total)), WinningMoney::sum);
+	}
+
+	public RankResult countRank(LottoRank rank) {
+		return lottoRanks.stream()
+			.filter(lottoRank -> Objects.equals(rank, lottoRank))
+			.collect(collectingAndThen(counting(), count -> new RankResult(rank, (int)(long) count)));
```

Comment:
```suggestion
			.collect(collectingAndThen(counting(), count -> new RankResult(rank, (int) count)));
```
명시적 타입 캐스팅을 한번만 해도 되지 않을까요?

Reply:
한번만 하면 컴파일 에러가 나네요